tools/stubgen/parse.py

```diff
@@ -0,0 +1,151 @@
+"""Parser for function docstrings."""
+import ast
+import inspect
+import re
+from typing import Any, List, Optional
+
+from .config import CPP_CORE_MODULE_NAME, squash_overloads
+from .transformer import AddOverloadedDecorator, DropFunctionBody, SetFunctionName, \
+    add_decorator
+
+
+def _is_overloaded(docstring: str) -> bool:
+    lines = docstring.split('\n')
+    if len(lines) > 1 and 'Overloaded' in lines[1]:
+        return True
+    return False
+
+
+def _fix_cpp_namespace_access(docstring: str) -> str:
+    """Some docstrings contain C++-style namespace accesses,
+    e.g. scipp::dataset::DataArray.
+    Convert those to {CORE_MODULE_NAME}.DataArray
+    """
+    return (docstring.replace("scipp::dataset::DataArray",
+                              f"{CPP_CORE_MODULE_NAME}.DataArray").replace(
+                                  "scipp::dataset::Dataset",
+                                  f"{CPP_CORE_MODULE_NAME}.Dataset"))
```

Comment:
I think this is because of the binding order. We bind variable before dataset and data array. So when functions like `Variable.bin` are bound, Pybind11 doesn't yet know `DataArray`. So it inserts the C++ type name instead of the Python name.
`scipp::variable::Variable` never shows up because the class is bound before any functions using it.